traffic-mirroring/mirroring.c

```diff
@@ -958,14 +958,13 @@ int main(int argc, char **argv)
                     log_err("Failed to convert destination address to network "
                             "address %s",
                             dst_addr[i].addr);
-                    ret = EXIT_FAILURE;
-                    free(dst_key);
+		    free(dst_key);
+                    return (EXIT_FAILURE);
                 }
                 if (bpf_map_update_elem(dst_fd, dst_key, &dst_val, 0) < 0) {
                     log_err("Failed to update destination endpoint bpf map");
                     perror("ERROR: bpf_map_update_elem");
                     ret = EXIT_FAILURE;
-                    free(dst_key);
```

Comment:
Why are we not returning here, in case of EXIT_FAILURE ?

Reply:
Done and tested